File: benchmark/c/opl/opl_emu_registers_effective_rate/opl_emu_registers_effective_rate.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <string.h>

#define opl_min(a,b) (((a)<(b))?(a):(b))

uint32_t opl_emu_registers_effective_rate(uint32_t rawrate, uint32_t ksr);

uint32_t opl_emu_registers_effective_rate(uint32_t rawrate, uint32_t ksr)
{
	return (rawrate == 0) ? 0 : opl_min(rawrate + ksr, 63);
}

File: translation_result/real_world_code/success/c/opl/opl_emu_registers_effective_rate/gpt4/BaseRepair/opl_emu_registers_effective_rate.rs

fn opl_min(a: u32, b: u32) -> u32 {
    if a < b { a } else { b }
}

fn opl_emu_registers_effective_rate(rawrate: u32, ksr: u32) -> u32 {
    if rawrate == 0 {
        0
    } else {
        opl_min(rawrate.wrapping_add(ksr), 63)
    }
}
